File: Course/Course1/week2_algorithmic_warmup/7_last_digit_of_the_sum_of_fibonacci_numbers_again/fibonacci_partial_sum.py
# Uses python3
import sys

def fibonacci_partial_sum_naive(from_, to):
    if to<=1:
        return to
    
    sum_ = 0
    current = 0
    next  = 1

    if(from_<=1):
        sum_ = 1

    for i in range(2,to + 1):
        current, next = next, current + next
        current,next = current%10,next%10
        if i >= from_:
            sum_ += next
            sum_ %= 10

    return sum_ % 10


if __name__ == '__main__':
    input = sys.stdin.read();
    from_, to = map(int, input.split())
    print(fibonacci_partial_sum_naive(from_, to))

File: Course/Course1/week2_algorithmic_warmup/7_last_digit_of_the_sum_of_fibonacci_numbers_again/fibonacci_partial_sum.cpp
#include <iostream>
#include <vector>
using std::vector;

unsigned long long get_fibonacci_partial_sum_naive(unsigned long long from, unsigned long long to) {
    unsigned long long sum = 0;

    unsigned long long current = 0;
    unsigned long long next  = 1;

    if(from<=1){
        sum = 1;
    }

    for (long long i = 2; i <= to; ++i) {
        new_current = next;
        next = next + current;
        current = new_current;

        if (i >= from) {
            sum += current;
        }
    }

    return sum % 10;
}

int main() {
    unsigned long long from, to;
    std::cin >> from >> to;
    std::cout << get_fibonacci_partial_sum_naive(from, to) << '\n';
}
